core/state_transition.go

```diff
@@ -99,14 +100,19 @@ func IntrinsicGas(data []byte, accessList types.AccessList, isContractCreation,
 			}
 		}
 		// Make sure we don't exceed uint64 for all data combinations
-		if (math.MaxUint64-gas)/params.TxDataNonZeroGas < nz {
-			return 0, vm.ErrOutOfGas
```

Comment:
This change is for preparation of EIP2028. The reason we need EIP2028 is as follows:
If EIP2028 is not enabled, the test `revert_reason.json` will not pass unless we change the test itself. Check this commit that change the test json itself: https://github.com/XinFinOrg/XDPoSChain/pull/558/commits/0a496c39d67e4f5feb825a3fa02c5f1c768c2e6c
So actually this code change does not necessary belong to this PR. Do you want me to undo it?

Reply:
Notice: consensus will be changed if we enable eip-2028

Reply:
Yes please let's remove that from this PR. It would be a separate discussion if EIP-2028 is useful for XDC.

Reply:
> Yes please let's remove that from this PR. It would be a separate discussion if EIP-2028 is useful for XDC.

I removed it and force pushed commits.

Reply:
> The consensus will be changed if we enable eip-2028. I can include it in the PR of EIP-1559 if you need it.

No hurry. At the time of eip2028, we need to remember to undo this commit: https://github.com/XinFinOrg/XDPoSChain/pull/558/commits/e1b7ba677f2a640f8af92e2b6a5377e945f56f57